test/NuGet.Core.Tests/NuGet.Configuration.Test/SearchTreeTests.cs

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Linq;
+using FluentAssertions;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Configuration.Test
+{
+    public class SearchTreeTests
+    {
+        [Fact]
+        public void SearchTree_WithOneSource()
+        {
+            // Arrange
+            using var mockBaseDirectory = TestDirectory.Create();
+            var configPath1 = Path.Combine(mockBaseDirectory, "NuGet.Config");
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageNamespaces>
+        <clear />
+        <packageSource key=""nuget.org"">
+            <namespace id=""stuff"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+            var settings = Settings.LoadSettingsGivenConfigPaths(new string[] { configPath1 });
+
+            // Act & Assert
+            var configuration = PackageNamespacesConfiguration.GetPackageNamespacesConfiguration(settings);
+            Assert.True(configuration.IsNamespacesEnabled);
+
+            configuration.Namespaces.Should().HaveCount(1);
+            var packageSourcesMatchFull = configuration.GetConfiguredPackageSources("stuff");
```

Comment:
Add a test for `stuff.something` and assert that it returns null.

Reply:
Added.

Reply:
https://github.com/NuGet/NuGet.Client/blob/61c52b9d23b1b26310372d0d2b3d326efeefb761/test/NuGet.Core.Tests/NuGet.Configuration.Test/SearchTreeTests.cs#L41-L42